### PR TITLE
add precommit-hook

### PR DESCRIPTION
**Pull request by blacdev**
add bash script and make file

## Description

Add precommit-hoo to repo
## Motivation and Context

This change is required to keep the repo and anyone else inline in terms of what is allowed and what isn't.
## How Has This Been Tested?

It was tested locally. Look at the image below to see that it works.
## Screenshots (if appropriate - Postman, etc):
image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: app/database.py [hunk 1]
```diff
@@ -1,47 +1,54 @@
...
-        db_engine = create_engine(DATABASE_URL, connect_args={"check_same_thread": False})
+    if db_type == "sqlite":
+        db_engine = create_engine(
+            database_url, connect_args={"check_same_thread": False}
+        )
     else:
-        db_engine = create_engine(DATABASE_URL)
+        db_engine = create_engine(database_url)
 
     return db_engine
```
Comment (zxenonx) [inline app/database.py:53]:
```
    if db_type == "sqlite":
        db_engine = create_engine(DATABASE_URL, connect_args={"check_same_thread": False})
        return create_engine(
            database_url, connect_args={"check_same_thread": False}
        )

   db_engine = create_engine(database_url)
   return db_engine
```

File: app/services/external.py [hunk 2]
```diff
@@ -73,41 +83,44 @@ def fake_authenticate_user(fake_token: str ="l3h5.34jb3,4mh346gv,34h63vk3j4h5k43
...
+            "assessment.update.all",
+            "assessment.delete.own",
+            "assessment.delete.all",
+        ],
     }
 
     return AuthenticateUser(**data)
 
+
 def fetch_assessment_questions(user_id, assessment_id: str, count: bool, db: Session):
```
Comment (zxenonx) [inline app/services/external.py:105]: type hints

File: app/services/external.py [hunk 3]
```diff
@@ -160,7 +173,7 @@ def fetch_assessment_questions(user_id, assessment_id: str, count: bool, db: Ses
     return questions, None
 
 
-def fetch_single_assessment( assessment_id: int, db: Session):
+def fetch_single_assessment(assessment_id: int, db: Session):
```
Comment (zxenonx) [inline app/services/external.py:176]: type hints

File: app/services/user_session.py [hunk 4]
```diff
@@ -110,46 +116,43 @@ def save_session(data: UserAssessmentanswer, user_id: int, db: Session, token: s
...
     }
-    response=requests.post(settings.MESSAGING,data=json.dumps(email_payload))
-    # if response.status_code != 200:
-    #     raise HTTPException(status_code=response.status_code,detail={"message":"Email Delivery Error"})
-    # return response.status_code
+    requests.post(settings.MESSAGING, data=json.dumps(email_payload))
 
-def assign_badge( assessment_id, token):
 
+def assign_badge(assessment_id, token):
```
Comment (zxenonx) [inline app/services/user_session.py:142]: type hints

File: app/utils.py [hunk 5]
```diff
@@ -1,8 +1,10 @@
 import uuid
+
+
 # Helper function to check if a string is a valid UUID
 def is_valid_uuid(uuid_string):
```
Comment (zxenonx) [inline app/utils.py:5]: type hint

File: script.py [hunk 6]
```diff
@@ -192,31 +135,29 @@ def run():
...
-  #       question_id=i.question_id,
-  #       answer_id=i.id,
-  #       selected_response=questions_and_answer_list[0].get("correct_answer")
-  #     )
-  #     db.add(user_response)
-  #     db.commit()
-  #     db.refresh(user_response)
-    
+    #   print("Done writing in database")
+    #   print(f"assessment_id: {assessment.id}")
```
Comment (zxenonx) [inline script.py:158]: can we remove this?

**Review by zxenonx — CHANGES_REQUESTED** [review]
Thanks @blacdev  
I left some comments

File: app/services/external.py (hunk 3 above)
Comment (blacdev) [reply app/services/external.py:176]: on which one here?

Comment (zxenonx) [reply app/services/external.py:178]: Oh sorry. 
My bad. It's alright

**Review by zxenonx — APPROVED** [review]
Good job @blacdev

**Pull request merged**